packages/core/admin/admin/src/pages/SettingsPage/pages/ApplicationInfosPage/components/LogoInput/tests/index.test.js

```diff
@@ -0,0 +1,288 @@
+          )
+        ).toBeInTheDocument()
+      );
+    });
+
+    it('should show error message when uploading unauthorized file size', async () => {
+      render({ initialStep: 'upload' });
+      const changeLogoButton = document.querySelector('button');
+      fireEvent.click(changeLogoButton);
+
+      const file = new File([new Blob(['1'.repeat(1024 * 1024 + 1)])], 'michka.png', {
+        type: 'image/png',
+      });
+
+      const fileInput = document.querySelector('[type="file"]');
+
+      fireEvent.change(fileInput, {
+        target: { files: [file] },
+      });
+
+      await waitFor(() =>
+        expect(
+          screen.getByText(
+            'The file uploaded is too large (max dimension: 750*750, max file size: 100KB)'
+          )
+        ).toBeInTheDocument()
+      );
+    });
+
+    it('should accept upload and lead user to next modal', async () => {
+      render({ initialStep: 'upload' });
+      const changeLogoButton = document.querySelector('button');
+      fireEvent.click(changeLogoButton);
+
+      const file = new File(['(⌐□_□)'], 'michka.png', { type: 'image/png' });
+
+      const fileInput = document.querySelector('[type="file"]');
+
+      fireEvent.change(fileInput, {
+        target: { files: [file] },
+      });
+
+      await waitFor(() => expect(screen.getByText('Pending logo')).toBeInTheDocument());
+    });
+
+    it('should let user choose another logo', async () => {
+      render({ initialStep: 'upload' });
+      const changeLogoButton = document.querySelector('button');
+      fireEvent.click(changeLogoButton);
+
+      const file = new File(['(⌐□_□)'], 'michka.png', { type: 'image/png' });
+
+      const fileInput = document.querySelector('[type="file"]');
+
+      fireEvent.change(fileInput, {
+        target: { files: [file] },
+      });
+
+      await waitFor(() => expect(screen.getByText('Pending logo')).toBeInTheDocument());
+
+      fireEvent.click(screen.getByText('Choose another logo'));
+
+      await waitFor(() => expect(screen.getByText('Upload logo')).toBeInTheDocument());
+    });
+  });
+
+  describe('from url', () => {
+    it('should render upload modal with from url tab', async () => {
+      render();
+      const changeLogoButton = document.querySelector('button');
+      fireEvent.click(changeLogoButton);
+
+      fireEvent.click(screen.getByText('From url'));
+
+      expect(document.body).toMatchSnapshot();
+    });
+
+    it('should show error message when uploading wrong file format', async () => {
+      render();
+      const changeLogoButton = document.querySelector('button');
+      fireEvent.click(changeLogoButton);
+      fireEvent.click(screen.getByText('From url'));
+
+      const textInput = document.querySelector('input[name="logo-url"]');
+
+      fireEvent.change(textInput, {
+        target: { value: 'https://thumbs.gfycat.com/AlertWarlikeBuzzard-size_restricted.gif' },
+      });
```

Comment:
Do you think we should rather use internal images here (e.g. something from strapi.io)? I have three issues with that:

- if the images from the remote source vanish the test fails
- we consume the bandwidth of an external service, which is not nice
- we expose ourselves to tracking of these external services

Reply:
That's a good point!
We need to find a file with a wrong format for this test though, that's why I used a gif 🤔